### PR TITLE
Support build index with multiple gpu

### PR DESCRIPTION
**Pull request by fishpenguin**
#346

### CODE REVIEW

**Review by wxyucs — APPROVED** [review]

**Review by BossZou — APPROVED** [review]

File: core/conf/server_gpu_config.template [hunk 1]
```diff
@@ -42,4 +42,5 @@ resource_config:
   search_resources:                 # define the devices used for search computation, must be in format: cpu or gpux
     - cpu
     - gpu0
-  index_build_device: gpu0          # CPU / GPU used for building index, must be in format: cpu or gpux
+  index_build_device:              # CPU / GPU used for building index, must be in format: cpu or gpux
```
Comment (cydrain) [inline core/conf/server_gpu_config.template]: index_build_resources:              # CPU or GPUs used for building index, must be in format: cpu or gpux

File: core/src/db/engine/ExecutionEngineImpl.cpp [hunk 2]
```diff
@@ -578,12 +578,19 @@ ExecutionEngineImpl::GpuCache(uint64_t gpu_id) {
 Status
 ExecutionEngineImpl::Init() {
     server::Config& config = server::Config::GetInstance();
-    Status s = config.GetResourceConfigIndexBuildDevice(gpu_num_);
+    std::vector<int64_t> gpu_ids;
```
Comment (cydrain) [inline core/src/db/engine/ExecutionEngineImpl.cpp]: use "device_ids" instead of "gpu_ids"

File: core/src/scheduler/SchedInst.cpp [hunk 3]
```diff
@@ -54,8 +54,8 @@ load_simple_config() {
     // get resources
     auto gpu_ids = get_gpu_pool();
 
-    int32_t build_gpu_id;
-    config.GetResourceConfigIndexBuildDevice(build_gpu_id);
+    std::vector<int64_t> build_gpu_ids;
```
Comment (cydrain) [inline core/src/scheduler/SchedInst.cpp]: use "index_build_device_ids"

File: core/src/server/Config.cpp [hunk 4]
```diff
@@ -215,8 +215,8 @@ Config::ValidateConfig() {
         return s;
     }
 
-    int32_t resource_index_build_device;
-    s = GetResourceConfigIndexBuildDevice(resource_index_build_device);
+    std::vector<int64_t> index_build_devices;
```
Comment (cydrain) [inline core/src/server/Config.cpp]: I'll update this file, please merge my CI before your commit.

**Pull request closed without merging**